src/Processors/Formats/Impl/MySQLDumpRowInputFormat.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <Processors/Formats/IRowInputFormat.h>
+#include <Processors/Formats/ISchemaReader.h>
+#include <Formats/FormatSettings.h>
+
+namespace DB
+{
+
+class MySQLDumpRowInputFormat final : public IRowInputFormat
+{
+public:
+    MySQLDumpRowInputFormat(ReadBuffer & in_, const Block & header_, Params params_, const FormatSettings & format_settings_);
+
+    String getName() const override { return "MySQLDumpRowInputFormat"; }
+    void readPrefix() override;
+
+    bool needContext() const override { return true; }
+    void setContext(ContextPtr & context_) override { context = Context::createCopy(context_); }
```

Comment:
Do we need context because we inference schema by interpreting `CREATE` queries? Or there's something else?
As I see only `settings.max_query_size` `settings.max_parser_depth` used from context, if so, maybe just set it to some reasonable default value not from context to get rid of dependency from it?